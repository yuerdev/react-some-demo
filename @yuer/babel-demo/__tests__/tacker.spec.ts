const { transformFileSync } = require('@babel/core');
import { dirname } from "path";
import { fileURLToPath } from "url";
import trackerPlugin from '../tracker';
const path = require('path');


const __dirname = dirname(fileURLToPath(import.meta.url));
describe("tracker", () => {
    it("auto tracker", () => {
        const { code } = transformFileSync(path.join(__dirname, './trackerSource.js'), {
            root: __dirname,
            parserOpts: {
                plugins: ['jsx']
            },
            plugins: [[trackerPlugin, {
                trackerPath: 'tracker'
            }]],
        });

        expect(code).toMatchSnapshot();
    })
})