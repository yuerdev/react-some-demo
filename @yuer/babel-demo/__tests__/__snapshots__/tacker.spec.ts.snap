// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tracker auto tracker 1`] = `
"import _objectObject from \\"tracker\\";
import aa from 'aa';
import * as bb from 'bb';
import { cc } from 'cc';
import 'dd';

function a() {
  _objectObject();

  console.log('aaa');
}

class B {
  bb() {
    _objectObject();

    return 'bbb';
  }

}

const c = () => {
  _objectObject();

  return 'ccc';
};

const d = function () {
  _objectObject();

  console.log('ddd');
};"
`;
